eessi_bot_software_layer.py

```diff
@@ -9,65 +9,81 @@
-from requests.structures import CaseInsensitiveDict
 
-import handlers
 from connections import github
 from tools import args, config
-from tools.logging import log, log_event
-
-
-def read_event_from_json(jsonfile):
-    """
-    Read event data from a json file.
-    """
-    req = namedtuple('Request', ['headers', 'json'])
-    with open(jsonfile, 'r') as jf:
-        event_data = json.load(jf)
-        req.headers = CaseInsensitiveDict(event_data['headers'])
-        req.json = event_data['json']
-    return req
-
-
-def handle_event(request):
-    """
-    Handle event
-    """
-    event_type = request.headers["X-GitHub-Event"]
-
-    event_handler = handlers.event_handlers.get(event_type)
-    if event_handler:
-        event_handler(request)
-    else:
-        log("Unsupported event type: %s" % event_type)
-        response_data = {'Unsupported event type': event_type}
-        response_object = json.dumps(response_data, default=lambda obj: obj.__dict__)
-        return flask.Response(response_object, status=400, mimetype='application/json')
-
-
-def create_app():
-    """
-    Create Flask app.
-    """
-
-    app = flask.Flask(__name__)
-
-    @app.route('/', methods=['POST'])
-    def main():
-        # verify_request(flask.request)
-        log_event(flask.request)
-        # handle_event(flask.request)
-        return ''
-
-    return app
+from tasks.build import build_easystack_from_pr
+
+from pyghee.lib import PyGHee, create_app, read_event_from_json
+from pyghee.utils import log
+
+class EESSIBotSoftwareLayer(PyGHee):
+    def handle_issue_comment_event(self, event_info, log_file=None):
+        """
+        Handle adding/removing of comment in issue or PR.
+        """
+        request_body = event_info['raw_request_body']
+        issue_url = request_body['issue']['url']
+        comment_author = request_body['comment']['user']['login']
+        comment_txt = request_body['comment']['body']
+        log("Comment posted in %s by @%s: %s" % (issue_url, comment_author, comment_txt))
+        log("issue_comment event handled!", log_file=log_file)
+
+
+    def handle_installation_event(self, event_info, log_file=None):
+        """
+        Handle installation of app.
+        """
+        request_body = event_info['raw_request_body']
+        user = request_body['sender']['login']
+        repo_name = request_body['repositories'][0]['full_name']
+        log("App installed by user %s into repository %s" % (user,repo_name))
+        log("installation event handled!", log_file=log_file)
+
+
+    def handle_pr_label_event(self, event_info, pr):
+        """
+        Handle adding of a label to a pull request.
+        """
+        log("PR labeled")
+
+
+    def handle_pr_opened_event(self, event_info, pr):
+        """
+        Handle opening of a pull request.
+        """
+        log("PR opened")
+        build_easystack_from_pr(pr, event_info)
+
+
+    def handle_pull_request_event(self, event_info, log_file=None):
```

Comment:
This looks like a (default) generic handler that could be moved into PyGHee, no?

It basically refines a PR event to something more fine-grained using the `action` field (PR opened, PR labeled, etc.)

To avoid confusion, I would name the methods triggered by this `handle_pull_request_<action>_event` (rather than using `_pr_`).

You can also avoid the `handlers` dict by using something like this:

```python
handler_name = 'handle_pull_request_%s_event' % action
if hasattr(self, handler_name):
    handler = getattr(self, handler_name)
    handler(event_info, pr)
else:
    log("No handler for PR action '%s'" % action)
```

For now, we can leave this as is, and clean it up later once PyGHee has been enhanced with a method to refine a `pull_request` event into more specific handlers.

Reply:
Just changed this as suggested.